src/Hazelcast.Net/NearCaching/NearCacheStatistics.cs

```diff
@@ -146,24 +147,35 @@ public void NotifyStaleRead()
+            private const string NearCacheDescriptorPrefix = "nearcache";
+            private const string NearCacheDescriptorDiscriminator = "name";
+            private readonly string _nearCacheName;
+
+            private MetricDescriptor<T> CreateDescriptor<T>(string metricName, MetricUnit unit)
+                => MetricDescriptor.Create<T>(NearCacheDescriptorPrefix, metricName, unit)
+                    .WithAttributePrefixes(NearCacheAttributePrefix, _nearCacheName) // override as 'nc.<name>.metric=value' for attributes
+                    .WithDiscriminator(NearCacheDescriptorDiscriminator, _nearCacheName);
+
+            public MetricDescriptors(string nearCacheName)
             {
-                var prefix = "nc" + EnumerableOfMetricsExtensions.NameSeparator + name.TrimStart('/');
-
-                CreationTime = new MetricDescriptor<long>(prefix, "creationTime");
-                Evictions = new MetricDescriptor<long>(prefix, "evictions");
-                Hits = new MetricDescriptor<long>(prefix, "hits");
-                Misses = new MetricDescriptor<long>(prefix, "misses");
-                EntryCount = new MetricDescriptor<long>(prefix, "ownedEntryCount");
-                Expirations = new MetricDescriptor<long>(prefix, "expirations");
-                Invalidations = new MetricDescriptor<long>(prefix, "invalidations");
-                InvalidationRequests = new MetricDescriptor<long>(prefix, "invalidationRequests");
-                OwnedEntryMemoryCost = new MetricDescriptor<long>(prefix, "ownedEntryMemoryCost");
-
-                LastPersistenceDuration = new MetricDescriptor<long>(prefix, "lastPersistenceDuration");
-                LastPersistenceKeyCount = new MetricDescriptor<long>(prefix, "lastPersistenceKeyCount");
-                LastPersistenceTime = new MetricDescriptor<long>(prefix, "lastPersistenceTime");
-                LastPersistenceWrittenBytes = new MetricDescriptor<long>(prefix, "lastPersistenceWrittenBytes");
+                _nearCacheName = nearCacheName.TrimStart('/');
+
+                CreationTime = CreateDescriptor<long>("creationTime", MetricUnit.Milliseconds);
+                Evictions = CreateDescriptor<long>("evictions", MetricUnit.Count);
+                Hits = CreateDescriptor<long>("hits", MetricUnit.Count);
+                Misses = CreateDescriptor<long>("misses", MetricUnit.Count);
+                EntryCount = CreateDescriptor<long>("ownedEntryCount", MetricUnit.Count);
+                Expirations = CreateDescriptor<long>("expirations", MetricUnit.Count);
+                Invalidations = CreateDescriptor<long>("invalidations", MetricUnit.Count);
+                InvalidationRequests = CreateDescriptor<long>("invalidationRequests", MetricUnit.Count);
+                OwnedEntryMemoryCost = CreateDescriptor<long>("ownedEntryMemoryCost", MetricUnit.Bytes);
+
+                // these exist in Java but not in Python?
```

Comment:
Java is the reference, right?

Reply:
In theory yes, but there are many places in Java where things have become obsolete... Like, I am not too sure what these metrics are, and not all clients send them. Again, the issue here is that there is no place I can look at to learn how these things work, and even though "java is the reference", in many places it would lead us to implement obsolete things in other clients just because Java does it...?